server/lib/chat/chat-service.test.ts

```diff
@@ -270,27 +294,20 @@ describe('chat/chat-service', () => {
         { type: ServerChatMessageType.JoinChannel },
         dbClient,
       )
+
+      await transactionPromise.resolve()
+
       expect(client2.publish).toHaveBeenCalledWith(getChannelPath(shieldBatteryChannel.id), {
         action: 'join2',
         user: user1,
-        message: {
-          id: joinUser1ShieldBatteryChannelMessage.msgId,
-          type: joinUser1ShieldBatteryChannelMessage.data.type,
-          channelId: joinUser1ShieldBatteryChannelMessage.channelId,
-          userId: joinUser1ShieldBatteryChannelMessage.userId,
-          time: Number(joinUser1ShieldBatteryChannelMessage.sent),
-        },
+        message: toJoinChannelMessageJson(joinUser1ShieldBatteryChannelMessage),
+      })
+      expect(client1.publish).toHaveBeenCalledWith(getChannelPath(shieldBatteryChannel.id), {
```

Comment:
This doesn't verify anything new from the last version you had. You need to clear the mock before resolving the deferred or they'll still have the previous list of calls, and thus a call before or after the deferred being resolved makes zero difference to this succeeding

Reply:
Would generally encourage you to make sure tests actually fail when they do the wrong thing, otherwise we end up with a super false sense of protection. Tests that do not test the thing they say they do are actually worse than no tests at all imo